asterius/src/Asterius/CodeGen.hs

```diff
@@ -1465,9 +1466,29 @@ marshalCmmInstr instr = case instr of
 marshalCmmBlockBody :: [GHC.CmmNode GHC.O GHC.O] -> CodeGen [Expression]
 marshalCmmBlockBody instrs = concat <$> for instrs marshalCmmInstr
 
+-- ----------------------------------------------------------------------------
+
+-- | Datatype to capture whether there is at least on CmmSwitch without a
+-- DEFAULT target within a Cmm function. TODO: Place appropriately.
+data ContainsDefault = WithDefault | WithoutDefault
```

Comment:
My brain needs an extra layer of indirection when processing this thing: `WithoutDefault` -> need to insert `__asterius_unreachable`, `WithDefault` -> do not insert `__asterius_unreachable`.

My suggestion is:

* Rename the datatype from `ContainsDefault` to `NeedsUnreachableBlock`.
* Don't roll the `Semigroup`/`Monoid` instances by hand. Make `NeedsUnreachableBlock` a `newtype` of `Bool`, and use `deriving via` to get the instances of `Data.Monoid.Any`. Things are more obvious this way; when combining a bunch of `NeedsUnreachableBlock`s, if any one of them is true, then the result is true.